plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -1,15 +1,26 @@
 module SamsonPipelines::StageConcern
-  # Return true if any stages in the pipeline are marked production
-  def production?
```

Comment:
changes made.  a test is currently failing as a result (validation as you mention).